packages/common-integration-tests/src/auth-client.ts

```diff
@@ -106,6 +134,118 @@ export function runAuthClientTests(
         MomentoErrorCode.AUTHENTICATION_ERROR
       );
     });
+
+    it('should support generating a superuser token when authenticated via a session token', async () => {
+      const generateResponse = await sessionTokenAuthClient.generateAuthToken(
+        SUPER_USER_PERMISSIONS,
+        ExpiresIn.seconds(1)
+      );
+      expect(generateResponse).toBeInstanceOf(GenerateAuthToken.Success);
+    });
+
+    it('should support generating an AllDataReadWrite token when authenticated via a session token', async () => {
+      const generateResponse = await sessionTokenAuthClient.generateAuthToken(
+        AllDataReadWrite,
+        ExpiresIn.seconds(1)
+      );
+      expect(generateResponse).toBeInstanceOf(GenerateAuthToken.Success);
+    });
+
+    it('should support generating a superuser token when authenticated via a v1 superuser token', async () => {
```

Comment:
ohhhhh wait a second. I think... we don't want this. Drat.

I think a v1 token can only generate another v1 token with less permissions 😓 . need to fix this on the server side probably.